python/eos/tasks.py

```diff
@@ -0,0 +1,196 @@
+# vim: set sw=4 sts=4 et tw=120 :
+
+# Copyright (c) 2020-2021 Danny van Dyk
+#
+# This file is part of the EOS project. EOS is free software;
+# you can redistribute it and/or modify it under the terms of the GNU General
+# Public License version 2, as published by the Free Software Foundation.
+#
+# EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+# Place, Suite 330, Boston, MA  02111-1307  USA
+
+import eos
+import logging
+import numpy as _np
+import os
+import pypmc
+
+def sample_mcmc(analysis_file, posterior, chain, base_directory='./', pre_N=150, preruns=3, N=1000, stride=5):
+    """
+    Samples from a named posterior PDF using Markov Chain Monte Carlo (MCMC) methods.
+
+    The output file will be stored in EOS_BASE_DIRECTORY/POSTERIOR/mcmc-IDX.
+
```

Comment:
What does IDX refer to?

Reply:
It's the chain index. Replaced it with ``CHAIN`` instead. Good catch.